src/tonic/api/snapshots_api.rs

```diff
@@ -128,3 +133,110 @@ impl Snapshots for SnapshotsService {
         }))
     }
 }
+
+#[derive(Clone)]
+pub struct ShardSnapshotsService {
+    toc: Arc<TableOfContent>,
+}
+
+impl ShardSnapshotsService {
+    pub fn new(toc: Arc<TableOfContent>) -> Self {
+        Self { toc }
+    }
+}
+
+#[async_trait]
+impl ShardSnapshots for ShardSnapshotsService {
+    async fn create_shard(
+        &self,
+        request: Request<CreateShardSnapshotRequest>,
+    ) -> Result<Response<CreateSnapshotResponse>, Status> {
+        let request = request.into_inner();
+        validate(&request)?;
```

Comment:
In internal services we use `validate_and_log(request.get_ref());` rather than `validate(...)?`:

```suggestion
        validate_and_log(request.get_ref());
```

For example: https://github.com/qdrant/qdrant/blob/4df28d250dce058202fd45c18ab11ff108d3e7c5/src/tonic/api/points_internal_api.rs#L41

Also applies to the other actions in this service.

Reply:
Fixed.